src/main/java/com/ntnghia/nghianguyenblog/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+package com.ntnghia.nghianguyenblog.service.impl;
+
+import com.ntnghia.nghianguyenblog.entity.User;
+import com.ntnghia.nghianguyenblog.exception.BadRequestException;
+import com.ntnghia.nghianguyenblog.exception.NotFoundException;
+import com.ntnghia.nghianguyenblog.repository.RoleRepository;
+import com.ntnghia.nghianguyenblog.repository.UserRepository;
+import com.ntnghia.nghianguyenblog.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    RoleRepository roleRepository;
+
+    @Autowired
+    PasswordEncoder encoder;
+
+    @Override
+    public List<User> getAll() {
+        return userRepository.findAll();
+    }
+
+    @Override
+    public User findById(int id) {
+        if (isIdExist(id)) return userRepository.findById(id).get();
+
+        throw new NotFoundException(String.format("User id %d not found", id));
+    }
+
+    @Override
+    public List<User> findByKeyword(String keyword) {
+        return userRepository.findByFirstNameContainsOrLastNameContainsOrEmailContains(keyword, keyword, keyword);
+    }
+
+    @Override
+    public User saveUser(User user) {
+        if (isEmailExist(user.getEmail())) {
+            throw new BadRequestException("User email exist");
```

Comment:
Viết câu đầy đủ e nhé: `This email address is already being used...`

Reply:
Done